sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/CallToAsyncMethodShouldNotBeBlocking.cs

```diff
@@ -100,6 +96,11 @@ private static void ReportOnViolation(SyntaxNodeAnalysisContext context)
                 return;
             }
 
+            if (simpleMemberAccess.IsInNameofCall(context.SemanticModel))
```

Comment:
Can we move this before `possibleMemberAccesses` to have a tiny performance optimization? IsIsNameofCall is cheap, getting symbol info is more expensive.

Reply:
IsInNameofCall needs the semantic model to retrieve the symbols so it's actually the same thing

Reply:
Maybe in another PR: IsNameof can be improved (changed to `=>` as single bool expression) and take semanticModel performance penalty AFTER `Identifier.ToString() == NameOfKeywordText` check. 

In this PR: Moving the condition upwards will prepare this rule for this small performance improvement.

Reply:
I've changed the `IsNameof` method and moved the `IsInNameofCall` call up.